From 8922b4ac3420f0461c33f4306ccde5b0dddedf7d Mon Sep 17 00:00:00 2001
From: Loc Nguyen <loc.nguyen.ks@renesas.com>
Date: Tue, 18 Dec 2018 13:47:48 +0700
Subject: [PATCH] supporting kernel version 4.19 and later for kernel module
 mmngrbuf

due to update in kernel to 4.19, modifying mmngr_buf_drv.c and mmngr_buf_private.h
-add <linux/mod_devicetable.h> to use struct of_device_id
-removing unsued parameter in dmabuf_attach
-remove .map_atomic and .unmap_atomic because they are no longer used

Signed-off-by: Loc Nguyen <loc.nguyen.ks@renesas.com>
---
 .../mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c   | 16 ++--------------
 .../files/mmngrbuf/include/mmngr_buf_private.h           |  5 +----
 2 files changed, 3 insertions(+), 18 deletions(-)

diff --git a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c
index 1bb7cb1..09a7c0b 100644
--- a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c
+++ b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c
@@ -66,6 +66,7 @@
 #include <linux/miscdevice.h>
 #include <linux/platform_device.h>
 #include <linux/dma-buf.h>
+#include <linux/mod_devicetable.h>
 
 #include "mmngr_buf_private.h"
 
@@ -222,7 +223,7 @@ static struct miscdevice misc = {
 	.fops		= &fops,
 };
 
-static int dmabuf_attach(struct dma_buf *buf, struct device *dev,
+static int dmabuf_attach(struct dma_buf *buf, 
 	struct dma_buf_attachment *attach)
 {
 	return 0;
@@ -279,17 +280,6 @@ static int dmabuf_end_cpu_access(struct dma_buf *buf,
 	return 0;
 }
 
-static void *dmabuf_map_atomic(struct dma_buf *buf, unsigned long page)
-{
-	return NULL;
-}
-
-static void dmabuf_unmap_atomic(struct dma_buf *buf, unsigned long page,
-				void *vaddr)
-{
-
-}
-
 static void *dmabuf_map(struct dma_buf *buf, unsigned long page)
 {
 	return NULL;
@@ -334,8 +324,6 @@ static const struct dma_buf_ops dmabuf_ops = {
 	.release = dmabuf_release,
 	.begin_cpu_access = dmabuf_begin_cpu_access,
 	.end_cpu_access = dmabuf_end_cpu_access,
-	.map_atomic = dmabuf_map_atomic,
-	.unmap_atomic = dmabuf_unmap_atomic,
 	.map = dmabuf_map,
 	.unmap = dmabuf_unmap,
 	.mmap = dmabuf_mmap,
diff --git a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h
index dfbea57..81770c6 100644
--- a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h
+++ b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h
@@ -104,7 +104,7 @@ static long compat_ioctl(struct file *file, unsigned int cmd,
 static int open(struct inode *inode, struct file *file);
 static int close(struct inode *inode, struct file *file);
 static long ioctl(struct file *file, unsigned int cmd, unsigned long arg);
-static int dmabuf_attach(struct dma_buf *buf, struct device *dev,
+static int dmabuf_attach(struct dma_buf *buf,
 			struct dma_buf_attachment *attach);
 static void dmabuf_detach(struct dma_buf *buf,
 			struct dma_buf_attachment *attach);
@@ -118,9 +118,6 @@ static int dmabuf_begin_cpu_access(struct dma_buf *buf,
 				enum dma_data_direction direction);
 static int dmabuf_end_cpu_access(struct dma_buf *buf,
 				enum dma_data_direction direction);
-static void *dmabuf_map_atomic(struct dma_buf *buf, unsigned long page);
-static void dmabuf_unmap_atomic(struct dma_buf *buf, unsigned long page,
-				void *vaddr);
 static void *dmabuf_map(struct dma_buf *buf, unsigned long page);
 static void dmabuf_unmap(struct dma_buf *buf, unsigned long page,
 			 void *vaddr);
-- 
2.7.4

